packages/contracts/contracts/libs/v0.8.x/minter-libs/SEALib.sol

```diff
@@ -0,0 +1,323 @@
+        uint64 endTime;
+        // The minimum percentage increase required for the next bid
+        // value is copied from SEAProjectConfig, where it is also a uint8
+        uint8 minBidIncrementPercentage;
+        // Whether or not the auction has been settled
+        bool settled;
+    }
+
+    /**
+     * @notice Updates the SEAProjectConfig to clear the current settings for
+     * future auctions on the project. Values are reset to their default
+     * initial values.
+     * Current auction details are not cleared or affected.
+     * @param _SEAProjectConfig SEAProjectConfig to update
+     */
+    function resetFutureAuctionDetails(
+        SEAProjectConfig storage _SEAProjectConfig
+    ) internal {
+        // reset future auction details
+        _SEAProjectConfig.auctionDurationSeconds = 0;
+        _SEAProjectConfig.minBidIncrementPercentage = 0;
+        _SEAProjectConfig.basePrice = 0;
+        _SEAProjectConfig.timestampStart = 0;
+    }
+
+    /**
+     * Overwrite the active auction for a project with a new auction.
+     * @dev This function is used to initialize a new auction. Care must be
+     * taken to ensure that the existing auction is fully complete and settled.
+     * @param _SEAProjectConfig SEAProjectConfig to update
+     * @param _targetTokenId token ID to create the auction for
+     * @param _bidAmount initial bid amount
+     * @param _bidder initial bidder's payable address
+     * @return endTime end time of the newly created auction
+     */
+    function OverwriteProjectActiveAuction(
+        SEAProjectConfig storage _SEAProjectConfig,
+        uint256 _targetTokenId,
+        uint256 _bidAmount,
+        address payable _bidder
+    ) internal returns (uint64 endTime) {
+        // calculate auction end time
+        endTime = (block.timestamp + _SEAProjectConfig.auctionDurationSeconds)
+            .toUint64();
+        // set active auction on SEAProjectConfig
+        _SEAProjectConfig.activeAuction = Auction({
+            tokenId: _targetTokenId,
+            currentBid: _bidAmount,
+            currentBidder: _bidder,
+            endTime: endTime,
+            minBidIncrementPercentage: _SEAProjectConfig
+                .minBidIncrementPercentage,
+            settled: false
+        });
+        return endTime;
+    }
+
+    /**
+     * @notice Updates the bid state of an Auction.
+     * Note that the auction's end time is extended if the bid is placed within
+     * the last `_timeBufferSeconds` of the auction.
+     * NOTE: This function does not check if the bid is valid. It is assumed
+     * that the bid has already been checked for validity by the caller.
+     * @param _auction Auction to update
+     * @param _timeBufferSeconds Time buffer to extend the auction to if the
+     * bid is placed within the last `_timeBufferSeconds` of the auction.
+     * @param _bidAmount bid amount
+     * @param _bidder bidder's payable address
+     */
+    function AuctionUpdateBid(
```

Comment:
Is there a reason we're uppercasing this method name?
```suggestion
    function auctionUpdateBid(
```

Reply:
updated in [90401a2](https://github.com/ArtBlocks/artblocks-contracts/pull/871/commits/90401a2dec350595832a59a8b2cd3daf92f5d779)